math/src/stableswap/math.rs

```diff
@@ -24,15 +27,19 @@ pub fn calculate_out_given_in<const N: u8, const N_Y: u8>(
 	if idx_in >= balances.len() || idx_out >= balances.len() {
 		return None;
 	}
-	let new_reserve_out = calculate_y_given_in::<N, N_Y>(amount_in, idx_in, idx_out, balances, amplification)?;
-	balances[idx_out].checked_sub(new_reserve_out)
+	let reserves = normalize_reserves(balances);
+	let amount_in = normalize_value(amount_in, balances[idx_in].decimals, TARGET_PRECISION, Rounding::Down);
```

Comment:
Doesn't the target precision depend on the assets involved?

Reply:
no, target precision is used to normalize all values to that precision.

Reply:
or basically yes, but the asset decimals is passed in and it takes it into account.